plugin/src/main/java/io/spine/tools/gradle/bootstrap/Extension.java

```diff
@@ -117,6 +136,51 @@ public JavaScriptExtension enableJavaScript() {
      */
     void disableJavaGeneration() {
         java.disableGeneration();
+        toggleJavaTasks(false);
+    }
+
+    private void toggleJavaTasks(boolean shouldEnable) {
+        this.javaEnabled = shouldEnable;
+        toggleCompileJavaTasks(shouldEnable);
+        toggleTransitiveProtos(shouldEnable);
+    }
+
+    /**
+     * If the {@code protobuf} configuration is present, toggles the transitivity according to the
+     * specified parameter.
+     *
+     * <p>Disabling transitivity leads to exclusion of {@code spine} and
+     * {@code com.google.protobuf} dependencies.
+     *
+     * @param shouldEnable whether the transitivity should be enabled
+     */
+    private void toggleTransitiveProtos(boolean shouldEnable) {
+        project.configurations(closure((ConfigurationContainer container) -> {
+            try {
+                @SuppressWarnings("DuplicateStringLiteralInspection" /* Different context. */)
+                Configuration protobuf = container.getByName("protobuf");
+                protobuf.setTransitive(shouldEnable);
+            } catch (UnknownConfigurationException ignored) {
+                /*
+                 * If the configuration could not be found then transitive dependencies can be
+                 * considered disabled.
+                 */
+            }
+        }));
+    }
+
+    /**
+     * Attempts to find and change the {@code enabled} flag of
+     * {@code compileJava} and {@code compileTestJava} tasks in the current project.
+     *
+     * <p>If such tasks could not be found in the project, performs no action.
+     */
+    private void toggleCompileJavaTasks(boolean shouldEnable) {
+        TaskContainer tasks = project.getTasks();
+        Optional.ofNullable(tasks.findByPath(COMPILE_JAVA))
+                .ifPresent(task -> task.setEnabled(shouldEnable));
+        Optional.ofNullable(tasks.findByPath(COMPILE_TEST_JAVA))
+                .ifPresent(task -> task.setEnabled(shouldEnable));
```

Comment:
Please use `if (<..> != null)` constructs instead. Using `Optional` locally for branching is considered a bad practice.